verify/helpers_test.go

```diff
@@ -0,0 +1,200 @@
+	})
+	err = setProfileStatus(client, ctx, ID, profileStatus)
+	if err != nil {
+		t.Errorf("setProfileStatus returned an error: %v", err)
+	}
+
+	userDoc, err = client.Collection("users").Doc(ID).Get(ctx)
+	if err != nil {
+		t.Errorf("Unable to fetch user document: %v", err)
+	}
+
+	assert.Equal(t, profileStatus, userDoc.Data()["profileStatus"])
+}
+
+func TestGetUserData(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	client := newFirestoreMockClient(ctx)
+	defer cancel()
+
+	testCases := []struct {
+		name          string
+		userId        string
+		profileURL    interface{}
+		profileStatus interface{}
+		chaincode     interface{}
+		expectedErr   error
+	}{
+		{
+			name:          "success",
+			userId:        "1",
+			profileURL:    "http://test.com",
+			profileStatus: "VERIFIED",
+			chaincode:     "abc123",
+			expectedErr:   nil,
+		},
+		{
+			name:          "missing profile url",
+			userId:        "2",
+			profileURL:    nil,
+			profileStatus: "BLOCKED",
+			chaincode:     "abc123",
+			expectedErr:   errors.New("profile url is not a string"),
+		},
+		{
+			name:          "missing chaincode",
+			userId:        "3",
+			profileURL:    "http://test.com",
+			profileStatus: "BLOCKED",
+			chaincode:     "",
+			expectedErr:   errors.New("chaincode is blocked"),
+		},
+		{
+			name:          "invalid chaincode",
+			userId:        "4",
+			profileURL:    "http://test.com",
+			profileStatus: "BLOCKED",
+			chaincode:     123,
+			expectedErr:   errors.New("chaincode is not a string"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			client.Collection("users").Doc(testCase.userId).Set(ctx, map[string]interface{}{
+				"chaincode":     testCase.chaincode,
+				"profileURL":    testCase.profileURL,
+				"profileStatus": testCase.profileStatus,
+			})
+
+			profileURL, profileStatus, chaincode, err := getUserData(client, ctx, testCase.userId)
+
+			assert.Equal(t, testCase.expectedErr, err)
+			if testCase.expectedErr == nil {
+				assert.Equal(t, testCase.profileURL.(string), profileURL)
+				assert.Equal(t, testCase.profileStatus.(string), profileStatus)
+				assert.Equal(t, testCase.chaincode.(string), chaincode)
+			}
+		})
+	}
+}
+
+func TestGetUserIdFromBody(t *testing.T) {
+	// valid request body
+	body := []byte(`{"userId": "123"}`)
+	expected := "123"
+
+	actual := getUserIdFromBody(body)
+
+	if actual != expected {
+		t.Errorf("getUserIdFromBody returned %v, expected %v", actual, expected)
+	}
+
+	// empty request body
+	body = []byte(``)
+	expected = ""
+
+	actual = getUserIdFromBody(body)
+
+	if actual != expected {
+		t.Errorf("getUserIdFromBody returned %v, expected %v", actual, expected)
+	}
+
+	// invalid request body
+	body = []byte(`{"invalidProperty": ""}`)
+	expected = ""
+
+	actual = getUserIdFromBody(body)
+
+	if actual != expected {
+		t.Errorf("getUserIdFromBody returned %v, expected %v", actual, expected)
+	}
+}
```

Comment:
i guess this should also throw an error if an invalid property is passed.

Reply:
It is not related to unmarshal but I am talking about API behaviour

Reply:
Please create a ticket for this as discussed

Reply:
**Created Ticket:** https://github.com/Real-Dev-Squad/identity-service/issues/145